save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/usersettings/Action.kt

```diff
@@ -0,0 +1,178 @@
+/**
+ * Utilities for cli args parsing
+ */
+
+@file:Suppress("FILE_NAME_MATCH_CLASS")
+
+package com.saveourtool.save.frontend.components.views.usersettings
+
+import com.saveourtool.save.frontend.components.modal.displayModalWithClick
+import com.saveourtool.save.frontend.utils.*
+import csstype.ClassName
+import org.w3c.fetch.Response
+import react.*
+import react.dom.html.ButtonType
+import react.dom.html.ReactHTML.button
+import react.dom.html.ReactHTML.div
+
+val actionButton: FC<ActionProps> = FC {props ->
+    val windowOpenness = useWindowOpenness()
+    val (displayTitle, setDisplayTitle) = useState(props.title)
+    val (displayMessage, setDisplayMessage) = useState(props.message)
+    val (isError, setError) = useState(false)
+
+    val action = useDeferredRequest {
+        val response = props.sendRequest(props.typeOfOperation)(this)
+        if (response.ok) {
+            props.onActionSuccess(props.typeOfOperation.isClickMode)
+        } else {
+            setDisplayTitle(props.errorTitle)
+            setDisplayMessage(response.unpackMessage())
+            setError(true)
+            windowOpenness.openWindow()
+        }
+    }
+
+    div {
+        button {
+            type = ButtonType.button
+            className = ClassName(props.classes)
+            props.buttonStyleBuilder(this)
+            onClick = {
+                setDisplayTitle(props.title)
+                setDisplayMessage(props.message)
+                windowOpenness.openWindow()
+            }
+        }
+    }
+
+    displayModalWithClick(
+        title = displayTitle,
+        message = displayMessage,
+        clickMessage = props.clickMessage,
+        isOpen = windowOpenness.isOpen(),
+        conditionClickIcon = props.conditionClick && !isError,
+        onCloseButtonPressed = windowOpenness.closeWindowAction(),
+        buttonBuilder = {
+            if (isError) {
+                buttonBuilder("Ok") {
+                    windowOpenness.closeWindow()
+                    setError(false)
+                }
+            } else {
+                props.modalButtons(action, windowOpenness, this)
+            }
+        },
+        changeClickMode = {
+            props.typeOfOperation.changeClickMode(it)
+        }
+    )
+}
+
+/**
+ * Button with modal for something action
+ *
+ * @return noting
+ */
+external interface ActionProps : Props {
+    /**
+     * Type of action
+     */
+    var typeOfOperation: TypeOfAction
```

Comment:
You should not care about type of action when creating universal component.
1. You have button that can be pressed.
2. After you press the button, the confirmation modal is shown.
3. After you confirm the action, the response from backend is shown.

If you want to support checkbox in some cases, just make something (e.g. checkbox name or checkbox onChange) nullable and that is all.

Reply:
DELETED